src/components/PendingTxsList/index.tsx

```diff
@@ -0,0 +1,191 @@
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { lastViewedSafe } from 'src/logic/currentSession/store/selectors'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import useOwnerSafes from 'src/logic/safe/hooks/useOwnerSafes'
+import { useTransactionType } from 'src/routes/safe/components/Transactions/TxList/hooks/useTransactionType'
+import { useKnownAddress } from 'src/routes/safe/components/Transactions/TxList/hooks/useKnownAddress'
+import { generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import { getChainById } from 'src/config'
+import { useAssetInfo } from 'src/routes/safe/components/Transactions/TxList/hooks/useAssetInfo'
+import { TxInfo } from 'src/routes/safe/components/Transactions/TxList/TxCollapsed'
+
+const MAX_TXS_DISPLAY = 5
+
+const PendingTxsList = (): ReactElement => {
+  const chainId = useSelector(currentChainId)
+  const defaultSafe = useSelector(lastViewedSafe) || undefined
+
+  const { isLoading: isLoadingOwnerSafes, ownerSafes } = useOwnerSafes()
+  const [loadingOwnerSafes, setLoadingOwnerSafes] = useState<boolean>()
+
+  // Selected Safe should come from the SideBar task when it's done
+  const [selectedSafe, setSelectedSafe] = useState<string>()
+
+  const [isFetchingPendingTxs, setIsFetchingPendingTxs] = useState<boolean>()
+  const [queuedTransactions, setQueuedTransactions] = useState<TransactionSummary[]>([])
+
+  useEffect(() => {
+    if (isLoadingOwnerSafes === undefined) return
+
+    if (ownerSafes[chainId]?.length) {
+      setSelectedSafe(ownerSafes[chainId][0])
+    } else {
+      setSelectedSafe(defaultSafe)
+    }
+    setLoadingOwnerSafes(false)
+  }, [chainId, defaultSafe, isLoadingOwnerSafes, ownerSafes, selectedSafe])
+
+  // Fetch txs awaiting confirmations
+  useEffect(() => {
+    if (selectedSafe === undefined) return
+
+    let isCurrent = true
+    const fetchQueuedTxs = async () => {
+      const { results } = await getTransactionQueue(GATEWAY_URL, chainId, checksumAddress(selectedSafe))
+      const txs = results
+        .filter((result) => isTransactionSummary(result))
+        .slice(0, MAX_TXS_DISPLAY)
+        .map((item: TransactionListItem) => (item as Transaction).transaction)
+      if (isCurrent) {
+        setQueuedTransactions(txs)
+        setIsFetchingPendingTxs(false)
+      }
+    }
+    setIsFetchingPendingTxs(true)
+    fetchQueuedTxs()
+
+    return () => {
+      isCurrent = false
+    }
+  }, [chainId, selectedSafe])
+
+  if (isFetchingPendingTxs) {
+    return (
+      <>
+        {Array.from(Array(MAX_TXS_DISPLAY).keys()).map((_, idx) => (
+          <TransactionSkeleton key={idx} />
+        ))}
+      </>
+    )
+  }
+
+  if (selectedSafe && !queuedTransactions.length && isFetchingPendingTxs === false) {
+    return <h3>This Safe has no pending transactions</h3>
+  }
+
+  if (!loadingOwnerSafes && !selectedSafe) {
+    return <h3>No Safes to display</h3>
+  }
+
+  if (!selectedSafe) return <></>
```

Comment:
Personal preference but I think returning `null` here would be a lot cleaner. It doesn't render a fragment.

Reply:
Hmm, fragments are not rendered in the DOM.
I would leave as it is unless I'm breaking our code standard.
As per your suggestion we also had to change the component return type to `ReactElement | null`.

Reply:
Sorry, I meant in the tree.

Again, it's personal preference but with the suggested return type you can see at a glance that the component may not return anything, whereas just `ReactElement` suggests there will always be something present in the DOM.